stdlib/char.gr

```diff
@@ -0,0 +1,54 @@
+export *
+
+// The minimum value of Unicode characters
+// Number
+let min = 0x0000
+
+// The maximum value of Unicode characters
+// Number
+let max = 0x10FFFF
+
+// Returns true if the given number is a valid Unicode scalar value.
+// @param n: Number - the value to check
+// @returns Bool
+let isValid = (n) => {
+  n >= min && (n <= 0xD7FF || n >= 0xE000) && n <= max
+}
+
+// Returns the Unicode code point for the character
+// @param char: Char - the input character
+// @returns Number
+import foreign wasm code : Char -> Number from 'stdlib-external/char'
+
+// Returns the Char for the given code point. Fails if the code point is invalid.
+// @param codePoint: Number - the Unicode code point
+// @returns Char
+import foreign wasm fromCode : Number -> Char from 'stdlib-external/char'
+
+// Returns the next valid Unicode character by code point. Fails if the input character is U+10FFFF.
+// @param char: Char - the input character
+// @returns Char
+let succ = (c) => {
+  let codePoint = code(c)
+  if (codePoint == max) {
+    fail 'no valid Unicode code point past U+10FFF'
+  } else if (codePoint == 0xD7FF) {
+    fromCode(0xE000)
+  } else {
+    fromCode(codePoint + 1)
+  }
+}
```

Comment:
Since these only fail in one pretty specific (and uncommon) case, it didn't really seem necessary to me. Feels like the kind of thing you would always just `Option.get` without doing any checking.

Reply:
Alright, let's make sure to document it and give the reasoning somewhere. I think that's especially important for contributor docs.

Reply:
Yeah, for sure.